packages/client/src/pages/example/example.tsx

```diff
@@ -0,0 +1,212 @@
+          />
+        </div>
+        <button onClick={handleLogin}>Login</button>
+      </div>
+      <div>
+        <h3>Signup (adds new user to db and sets them in the store)</h3>
+        <div>
+          <label>username</label>
+          <input
+            value={signupUsername}
+            onChange={(e) => setSignupUsername(e.target.value)}
+          />
+        </div>
+        <div>
+          <label>password</label>
+          <input
+            value={signupPassword}
+            onChange={(e) => setSignupPassword(e.target.value)}
+          />
+        </div>
+        <div>
+          <label>email</label>
+          <input
+            value={signupEmail}
+            onChange={(e) => setSignupEmail(e.target.value)}
+          />
+        </div>
+        <button onClick={handleSignup}>Signup</button>
+      </div>
+      <div>
+        <h3>Current User in store</h3>
+        {user.id ? (
+          <div>
+            <p>id: {user.id}</p>
+            <p>username: {user.username}</p>
+            <p>email: {user.email}</p>
+          </div>
+        ) : (
+          <div>NONE</div>
+        )}
+      </div>
+    </div>
+  );
+};
+
+const PostExample = () => {
+  // misc local component state
+  const [title, setTitle] = useState("");
+  const [tag, setTag] = useState("");
+  const [bodyText, setBodyText] = useState("");
+
+  // store state + action dispatcher
+  const dispatch = useAppDispatch();
+  const locationFilter = useAppSelector((state) => state.post.locationFilter);
+  const user = useAppSelector((state) => state.user);
+  const posts = useAppSelector((state) =>
+    // here we're deriving a computed value from location + posts state.
+    // When either of those states change, this value will recompute.
+    state.post.posts.filter(
+      (post) => post.location === state.post.locationFilter,
+    ),
+  );
+
+  // fetch all posts on initial render (in reality we prob won't fetch ALL posts, just ones in curr location)
+  useEffect(() => {
+    dispatch(getPosts());
+  }, [dispatch]);
+
+  const handleCreatePost = (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    if (user.id) {
+      dispatch(
+        createPost({
+          title,
+          bodyText,
+          tag,
+          location: locationFilter,
+          userID: user.id,
+        }),
+      )
+        .then(unwrapResult)
+        .then((data) => alert("post created oooo"))
+        .catch((error) => {
+          alert("post creation failed NOOOOO");
+        });
+    } else {
+      alert("you must be logged in to create a post");
+    }
+  };
+
+  return (
+    <div>
+      <h2>Post example</h2>
+      <select onChange={(e) => dispatch(setLocationFilter(e.target.value))}>
+        <option value="vancouver">Vancouver</option>
+        <option value="surrey">Surrey</option>
+        <option value="richmond">Richmond</option>
+        <option value="vancouver">Yellowknife</option>
```

Comment:
y is the value vancouver but the text yellowknife this is [UNACCEPTABLE](https://www.youtube.com/watch?v=07So_lJQyqw)

Reply:
jk

Reply:
ppl in yellowknife want to see vancouver feed 🍋